File: nos2jdbc-tutorial-spring/src/main/java/nos2jdbc/tutorial/spring/service/MemberService.java
package nos2jdbc.tutorial.spring.service;

import java.util.List;

import nos2jdbc.tutorial.spring.entity.Member;
import nos2jdbc.tutorial.spring.gen.service.MemberServiceBase;

import static nos2jdbc.tutorial.spring.gen.names.MemberNames.*;
import static org.seasar.extension.jdbc.operation.Operations.*;

import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@Service
@Transactional
public class MemberService extends MemberServiceBase {
    public List<Member> findAllWithClubs() {
	return select()
		.leftOuterJoin(clubMemberRelList())
		.leftOuterJoin(clubMemberRelList().club())
		.getResultList();
    }

    public Member getByOrder(int mo) {
	return select().orderBy(asc(id())).offset(mo).limit(1).getSingleResult();
    }
}

File: nos2jdbc-tutorial-kotlin-spring/src/main/kotlin/nos2jdbc/tutorial/kotlinspring/service/MemberService.kt
package nos2jdbc.tutorial.kotlinspring.service

import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional
import nos2jdbc.tutorial.kotlinspring.entity.Member
import nos2jdbc.tutorial.kotlinspring.gen.service.MemberServiceBase

import nos2jdbc.tutorial.kotlinspring.gen.names.MemberNames.*
import org.seasar.extension.jdbc.operation.Operations.*


@Service
@Transactional
class MemberService(): MemberServiceBase() {
    public fun findAllWithClubs(): List<Member> {
	return select()
		.leftOuterJoin(clubMemberRelList())
		.leftOuterJoin(clubMemberRelList().club())
		.getResultList();
    }

    public fun getByOrder(mo: Int): Member {
	return select().orderBy(asc(id())).offset(mo).limit(1).getSingleResult();
    }
}